lib/pbench/server/api/resources/__init__.py

```diff
@@ -41,9 +42,38 @@ def __str__(self) -> str:
         return self.message
 
 
-class UnauthorizedAccess(APIAbort):
+class APIInternalError(APIAbort):
+    """Used to report a server internal error with a UUID value that connects
+    the string reported to the client with a server log entry to aid analysis
+    by an SRE
     """
-    The user is not authorized for the requested operation on the specified
+
+    def __init__(self, logger: Logger, message: str):
+        """Construct an "internal server error" exception object.
+
+        This exception is raised and will be used in the API _dispatch method
+        to generate a JSON error message payload to the client. This message
+        contains a UUID string that uniquely identifies this internal server
+        error context. An internal server log message is generated, with
+        traceback, also containing that UUID value so that a server
+        administrator can investigate the cause of the error.
+
+        NOTE: We want to return minimal explanations of internal errors to the
+        client while capturing detailed information for server developers to
+        determine what happened.
+
+        Args:
+            logger  Python logger
+            message A brief descriptive message for the client and log
+        """
+        self.uuid = uuid.uuid4()
+        msg = f"INTERNAL error {message!r}: report error code {self.uuid} to an SRE"
+        super().__init__(http_status=HTTPStatus.INTERNAL_SERVER_ERROR, message=msg)
```

Comment:
The only likely danger would be if the super-class added a `uuid` attribute which it then initialized to a different value after the sub-class one is captured by the `msg` text.

Reply:
If we ever did that, we'd need to consciously decide which `uuid` value we wanted anyway; I wouldn't count on either alternative here necessarily being the "right" one in that context ... so that scenario really doesn't bother me.

Reply:
Yeah, well, late-binding would help:  if we didn't capture the UUID value in the message text until we were ready to use the text, then having the super-class initialization overwriting it wouldn't cause an issue.

Nevertheless, if we performed the initialization in the proper order (i.e., superclass first, then the immediate class), there would be fewer surprises all around, even if the superclass is later changed.